dataconnect/dataconnect/connector/queries.gql

```diff
@@ -0,0 +1,576 @@
+  movies(where: { title: { contains: $input } }) {
+    id
+    title
+    genre
+    rating
+    imageUrl
+  }
+}
+
+# Fetch a single movie using key scalars (same as get movie by id)
+query MovieByKey($key: Movie_Key!) @auth(level: PUBLIC) {
+  movie(key: $key) {
+    title
+    imageUrl
+  }
+}
+
+# Fetch movies by title
+query MovieByTitle($title: String!) @auth(level: PUBLIC) {
+  movies(where: { title: { eq: $title } }) {
+    id
+    title
+    imageUrl
+    genre
+    rating
+  }
+}
+
+# Fetch top-rated movies by genre
+query MovieByTopRating($genre: String) @auth(level: PUBLIC) {
+  mostPopular: movies(
+    where: { genre: { eq: $genre } }
+    orderBy: { rating: DESC }
+  ) {
+    id
+    title
+    imageUrl
+    rating
+    tags
+  }
+}
+
+# List movies by tag
+query ListMoviesByTag($tag: String!) @auth(level: PUBLIC) {
+  movies(where: { tags: { includes: $tag } }) {
+    id
+    title
+    imageUrl
+    genre
+    rating
+  }
+}
+
+# List top 10 movies
+query MoviesTop10 @auth(level: PUBLIC) {
+  movies(orderBy: [{ rating: DESC }], limit: 10) {
+    id
+    title
+    imageUrl
+    rating
+    genre
+    tags
+    metadata: movieMetadatas_on_movie {
+      director
+    }
+    mainActors: actors_via_MovieActor(where: { role: { eq: "main" } }) {
+      id
+      name
+      imageUrl
+    }
+    supportingActors: actors_via_MovieActor(where: { role: { eq: "supporting" } }) {
+      id
+      name
+      imageUrl
+    }
+  }
+}
+
+# List movies by release year range
+query MoviesByReleaseYear($min: Int, $max: Int) @auth(level: PUBLIC) {
+  movies(
+    where: { releaseYear: { le: $max, ge: $min } }
+    orderBy: [{ releaseYear: ASC }]
+  ) {
+    id
+    rating
+    title
+    imageUrl
+  }
+}
+
+# List recently released movies
+query MoviesRecentlyReleased @auth(level: PUBLIC) {
+  movies(where: { releaseYear: { ge: 2010 } }) {
+    id
+    title
+    rating
+    imageUrl
+    genre
+    tags
+  }
+}
+
+# List movies with filtering on fields
+query ListMoviesFilter($genre: String, $limit: Int) @auth(level: PUBLIC) {
+  movies(where: { genre: { eq: $genre } }, limit: $limit) {
+    title
+    imageUrl
+  }
+}
+
+# List movies by partial title string match
+query ListMoviesByTitleString(
+  $prefix: String
+  $suffix: String
+  $contained: String
+) @auth(level: PUBLIC) {
+  prefixed: movies(where: { description: { startsWith: $prefix } }) {
+    title
+  }
+  suffixed: movies(where: { description: { endsWith: $suffix } }) {
+    title
+  }
+  contained: movies(where: { description: { contains: $contained } }) {
+    title
+  }
+}
+
+# List movies by rating and genre with OR/AND filters
+query ListMoviesByRatingAndGenre($minRating: Float!, $genre: String)
+@auth(level: PUBLIC) {
+  movies(
+    where: { _or: [{ rating: { ge: $minRating } }, { genre: { eq: $genre } }] }
+  ) {
+    title
+    imageUrl
+  }
+}
+
+# Get favorite movies by user ID
+query GetFavoriteMoviesById($id: String!) @auth(level: USER) {
+  user(id: $id) {
```

Comment:
Please use `id_expr` to make it secure.

It should work now.

Same for other lookup below.

Reply:
Not done yet~ Though out this page. Security matters

Let me know `$id` here is supposed to match the Firebase Auth UID.